core/baseline-core-v1.0-psd01.md

```diff
@@ -3001,60 +3003,227 @@ This section of the document focuses on the requirements and considerations rela
+8. Complete the workstep using the agreed upon external authoritative data.
+9. Verify that the workstep was successfully completed using the agreed upon external authoritative data.
+
+Test Passing Criteria: The test is considered passing if,
+
+* The BPI system successfully initiates the agreement process among the BPI subjects.
+* The specified quorum of BPI subjects agree upon the source and type of the external authoritative data.
+* The BPI system records the agreement and uses the agreed upon external authoritative data as the input source for the workstep.
+* The workstep is successfully completed using the agreed upon external authoritative data.
 
 ## 8.3 External Non-authoritative, Non-deterministic Data for BPIs
 
+###### extnonauthbpidata
+
 External non-authoritative, non-deterministic data means that there does not exist an authoritative, deterministic source for the data used as input to a BPI workstep. In that case, a normalizing mechanism often referred to as an oracle is needed to account for potential source discrepancies. Entities reporting external non-authoritative, non-deterministic data into a BPI worksteps may also be referred to as oracles.
 
 ### 8.3.1 Data Trustworthiness
 
+###### bpidatatrust
+
 External non-authoritative, non-deterministic data for BPIs resides outside of an authoritative source and is subject to manipulation. As such, steps should be taken to remove counterparty manipulation and error risk through mechanisms such as redundancy in data reporting and error checking, either cross-party, or by an appropriately incentivized 3rd party.
 
 #### **[D42]** 
-External non-authoritative, non-deterministic BPI input data into a BPI workstep SHOULD be sourced from multiple endpoints.
+External non-authoritative, non-deterministic BPI input data into a BPI workstep SHOULD be sourced from multiple sources.
 
-[[D42]](#d42) Testability: Since the [Baseline Protocol API specification is testable](https://github.com/eea-oasis/baseline-standard/blob/main/api/baseline-api-v1.0-psd01.md), and since non-authoritative, non-deterministic data with source identification will be used in the Baseline Protocol APIs based on agreed upon data schames between counterparties, the requirement is testable.
+[[D42]](#d42) Testability: 
 
-#### **[R316]** 
+
+Preconditions:
+
+* A test BPI system is available and functional.
+* At least two independent sources of external non-authoritative, non-deterministic are available that produce the same input data.
+* A test workstep has been created that requires external non-authoritative, non-deterministic input data.
+
+Test Steps:
+
+1. Start the test workstep that requires external non-authoritative, non-deterministic input data.
+2. Source the input data from a single external source and record the source of the data.
+3. Repeat step 2, but use a different external source for the same input data.
+4. Repeat step 2, alternating between the external sources used in steps 2 and 3.
```

Comment:
How often? (Is this necessary at all?)

Reply:
step 4 is not necessary and I corrected step 3 to "for each available  external source ..."